### PR TITLE
[manila] add share_group_id to share's CreateOpts

### PR DESCRIPTION
**Pull request by kayrus**
Partially fixes the #3204

### CODE REVIEW

**Review by EmilienM — DISMISSED** [review]

**Comment by kayrus** [comment]
@EmilienM can you approve this?

**Comment by kayrus** [comment]
@pierreprinetti @mdbooth @EmilienM kindly ping

File: openstack/sharedfilesystems/v2/shares/requests.go [hunk 1]
```diff
@@ -40,6 +40,8 @@ type CreateOpts struct {
 	SnapshotID string `json:"snapshot_id,omitempty"`
 	// Determines whether or not the share is public
 	IsPublic *bool `json:"is_public,omitempty"`
+	// The UUID of the share group. Available starting from the microversion 2.31
+	ShareGroupID string `json:"share_group_id,omitempty"`
```
Comment (mdbooth) [inline openstack/sharedfilesystems/v2/shares/requests.go:44]: https://docs.openstack.org/api-ref/shared-file-system/#create-share ✅

File: openstack/sharedfilesystems/v2/shares/results.go [hunk 2]
```diff
@@ -54,6 +54,8 @@ type Share struct {
 	ShareType string `json:"share_type"`
 	// The name of the share type.
 	ShareTypeName string `json:"share_type_name"`
+	// The UUID of the share group. Available starting from the microversion 2.31
+	ShareGroupID string `json:"share_group_id"`
```
Comment (mdbooth) [inline openstack/sharedfilesystems/v2/shares/results.go:58]: https://docs.openstack.org/api-ref/shared-file-system/#create-share ✅

**Review by mdbooth — COMMENTED** [review]
Marshalling looks correct. Does this need tests?

/lgtm

**Review by EmilienM — APPROVED** [review]
can't we have tests?

**Comment by kayrus** [comment]
> can't we have tests?

in order to have tests we need to implement a `sharegroups` package. we don't have it yet. so far this is more useful if you create a share group once using openstack CLI.

**Pull request merged**